### ISSUE TITLE
[MPS] layer_norm forward kernel variance computation causes divergence from CPU reference

### ISSUE THREAD

**Issue by jhavukainen** [pytorch/pytorch#190491]
### 🐛 Describe the bug

The MPS forward kernel computes variance in one pass as `E[x²] − E[x]²` and then clamps `var < 1e-6` to zero. For a row with nearly identical values, the two quantities become nearly equal (and possibly large), so the small true variance is lost to rounding noise. Clamping it to zero below a certain level is just a band-aid to avoid the unrealistic case of negative variance, with the negative side effect of also setting valid positive but small variances to zero. This collapses reciprocal of standard deviation to `rsqrt(eps)` and makes the normalized outputs diverge between MPS backend and CPU.

Repro for the issue from 
```python
import torch
import torch.nn.functional as F

# Per-element variance ~9.8e-7 just below the kernel's 1e-6 clamp
# with the eps that ViT-family LayerNorms use (1e-12).
x = torch.tensor([[-9.9e-4, -9.9e-4, 9.9e-4, 9.9e-4]])
print(F.layer_norm(x,            (4,), eps=1e-12))   # CPU
print(F.layer_norm(x.to('mps'),  (4,), eps=1e-12))   # MPS

# Output:
# tensor([[-1.0000, -1.0000,  1.0000,  1.0000]])
# tensor([[-990., -990.,  990.,  990.]], device='mps:0')
```

The fix to avoid the cancellation is to compute the subtraction before the square by using the other identity for variance `E[(x−μ)²]`. This requires rewriting the kernel in a two-pass formulation, but also removes the need for the clamping as we can no longer get negative variances due to the cancellation.

Opening a PR with the suggested changes.

### Versions

Current nightly. MacOS27 beta.

cc @kulinseth @malfet @DenisVieriu97 @aditvenk @Isalia20
